versioned/tiered/rocksdb/src/main/java/com/dremio/nessie/versioned/store/rocksdb/RocksL3.java

```diff
@@ -0,0 +1,96 @@
+/*
+ * Copyright (C) 2020 Dremio
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.dremio.nessie.versioned.store.rocksdb;
+
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+import com.dremio.nessie.tiered.builder.L3;
+import com.dremio.nessie.versioned.impl.condition.ExpressionPath;
+import com.dremio.nessie.versioned.store.Id;
+import com.dremio.nessie.versioned.store.KeyDelta;
+import com.dremio.nessie.versioned.store.StoreException;
+import com.google.protobuf.InvalidProtocolBufferException;
+
+class RocksL3 extends RocksBaseValue<L3> implements L3, Evaluator {
+  private static final String TREE = "tree";
+
+  private Stream<KeyDelta> keyDelta;
+
+  RocksL3() {
+    super();
+  }
+
+  @Override
+  public L3 keyDelta(Stream<KeyDelta> keyDelta) {
+    this.keyDelta = keyDelta;
+    return this;
+  }
+
+  @Override
+  public boolean evaluate(Condition condition) {
+    for (Function function: condition.getFunctionList()) {
+      // Retrieve entity at function.path
+      if (function.getPath().getRoot().isName()) {
+        ExpressionPath.NameSegment nameSegment = function.getPath().getRoot().asName();
+        final String segment = nameSegment.getName();
+        if (segment.equals(ID)) {
+          if (!(!nameSegment.getChild().isPresent()
+              && function.getOperator().equals(Function.EQUALS)
+              && getId().toEntity().equals(function.getValue()))) {
+            return false;
+          }
+        } else {
+          // TODO: ConditionExpression is currently not supported for TREE.
+          // TODO: Implement a case for TREE if serialization of a Stream of KeyDelta to Entity is supported.
+          // Invalid Condition Function.
+          return false;
```

Comment:
Shouldn't this throw an UnsupportedException?

Reply:
Good point but I'm going to remove this TODO. I think evaluate() should behave the same for all tests that don't pass by returning false. Otherwise we should throw an exception, for instance, when a NameSegment in the ConditionException does not match an attribute in the object on which we run the evaluate().